doc/wg/network/notes/network-notes-2023-08-10.md

```diff
@@ -0,0 +1,96 @@
+ * Alex: Professor at Polytechnic university in Bucharest. Started on Tock in 2020, interested in Rust OS for teaching. Still interested in network communication for teaching.
+
+# Organization of meetings
+ * Alex: Let's begin with organization. I say the same as Core WG model: someone takes notes and posts to github. Markdown.
+ * Leon: One thing that bugs me from the Core call is that when you volunteer notes you end up having to multitask and do notes later. Maybe we could have a way to share the notes in real time, so someone else can take over
+ * Alex: That sounds good to me. What platform?
+ * Leon: Tool called HackMD
+ * Branden: I vote for that for next meeting
+ * Branden: How do we handle agenda? Should we send a notification in advance, or start the call with agenda. Or end with agenda?
+ * Leon: We can play this loose. I do like having the call for items as a reminder
+ * Branden: Is sending out the agenda reminder on Slack fine?
+ * Alex: We'll need to make sure everyone is on it
+ * Alex: What about drawings? Need some collaborative platform for that.
+ * Tyler: Generally in Tock, the documentation in terms of diagrams is pretty lacking in Tock. I think a great goal would be to ensure that we create visual documentation as part of our working group.
+ * Alex: I used draw.io in the past
+ * Branden: Alex has an ACTION ITEM to figure out a drawing platform
+
+# Working Group PR
+ * Branden: https://github.com/tock/tock/pull/3578 needs two more small edits before merging
+ * Alex: I will update that tonight
+ * Branden: We can merge tomorrow on the core team call
+
+# Overview of Network Stacks
+ * Leon: Would be good to do a recap of what people on this call have worked on. Then we could have visualizations of those for next week.
+ * Alex: CAN, Ethernet, Thread all seem in progress
+
+## Ethernet
+ * Leon: Trying to start with Ethernet. Been working on this for a long time. We have discussed at several past meetings of Tock groups. We specifically chose Ethernet as an example because it's one of the simplest transports to integrate with existing IP networks. Back in 2022 we didn't have any layer2 transport implementation. Over the last year we have written three proper implementations of Ethernet MAC layer. They differ in feature set, APIs, buffer management. So a key takeaway now is defining higher-level interfaces and going above layer2 in the OSI stack. IP layer, UDP transport, ultimately UDP/TCP based applications. We have a work-in-progress branch in Tock and Libtock-C where the Tock kernel has a Ethernet TAP driver (TAP/TUN from BSD Linux) which is a driver which forwards Ethernet layer2 to userspace and back. And we have userspace libraries to implement IP: LWIC. We're at the stage in the work-in-progress environment where we can share a small HTTP webpage. At Tockworld 2022 we had decided to start with Ethernet layer2. It's a great place to start, but definitely not what we want in the long run.
```

Comment:
```suggestion
 * Leon: Trying to start with Ethernet. Been working on this for a long time. We have discussed at several past meetings of Tock groups. We specifically chose Ethernet as an example because it's one of the simplest transports to integrate with existing IP networks. Back in 2022 we didn't have any layer2 transport implementation. Over the last year we have written three proper implementations of Ethernet MAC layer. They differ in feature set, APIs, buffer management. So a key takeaway now is defining higher-level interfaces and going above layer2 in the OSI stack. IP layer, UDP transport, ultimately UDP/TCP based applications. We have a work-in-progress branch in Tock and Libtock-C where the Tock kernel has a Ethernet TAP driver (TAP/TUN from BSD Linux) which is a driver which forwards Ethernet layer2 to userspace and back. And we have userspace libraries to implement IP: LwIP and `smoltcp`. We're at the stage in the work-in-progress environment where we can share a small HTTP webpage. At Tockworld 2022 we had decided to start with Ethernet layer2. It's a great place to start, but definitely not what we want in the long run.
```

Reply:
For me, GitHub fails to highlight the diff here. I propose changing "LWIC" to "LwIP and `smoltcp`".